threads-web-ui/app/[username]/page.tsx

```diff
@@ -0,0 +1,77 @@
+import Image from 'next/image';
+import React from 'react';
+import { Thread, ThreadsIcons } from 'react-threads';
+import { ThreadsAPI } from 'threads-api';
+
+const UserProfilePage = async ({ params }: { params: { username: string } }) => {
+  const threadsAPI = new ThreadsAPI();
+  const username = params.username;
+
+  if (!username) {
+    return;
+  }
+
+  const userID = await threadsAPI.getUserIDfromUsername(username);
+  if (!userID) {
+    return <div>No user with the given username</div>;
+  }
+
+  const userInfo: [string, string] = [username, userID];
+
+  const [userProfile, userThreads] = await Promise.all([
+    threadsAPI.getUserProfile(...userInfo),
+    threadsAPI.getUserProfileThreads(...userInfo),
+  ]);
```

Comment:
- Super nit I still think it should be `params`
- But this is unnecessary anymore, I changed the interface in #121, so we can just change `userID` now. 😁 
- So I'll just pass this. 👍